cloudinary_cli/utils/api_utils.py

```diff
@@ -116,18 +117,20 @@ def upload_file(file_path, options, uploaded=None, failed=None):
     verbose = logger.getEffectiveLevel() < logging.INFO
 
     try:
-        size = path.getsize(file_path)
         upload_func = uploader.upload
-        if size > 20000000:
-            upload_func = uploader.upload_large
+        if not re.match(r'^https?://', str(file_path)):
+            size = path.getsize(file_path)
+            if size > 20000000:
+                upload_func = uploader.upload_large
         result = upload_func(file_path, **options)
         disp_path = _display_path(result)
         disp_str = f"as {result['public_id']}" if not disp_path \
             else f"as {disp_path} with public_id: {result['public_id']}"
         logger.info(style(f"Successfully uploaded {file_path} {disp_str}", fg="green"))
         if verbose:
             print_json(result)
-        uploaded[file_path] = {"path": asset_source(result), "display_path": disp_path}
+        if "async" not in options:
```

Comment:
I had to update the last line to `asset_details.get('format', '')` as non-raw files still get the public_id as we specify them so this will prevent `no format` error as the response won't contain the format

Also I updated the display message to say it's processing when uploading asynchronously.